### DISCUSSION TITLE
Documentation - Variables

### DISCUSSION THREAD

**Discussion by Techbrunch** [projectdiscovery/nuclei#722] [Q&A] [answered]
Hi, is there a list of all the variable that can be used in a template such as {{Hostname}} and {{BaseURL}}, i need the protocol (http or https) and I think the port might be useful in some cases.

**Comment by ehsandeep** [comment, marked as answer]
Those are the ones currently nuclei supports. 

GitHub Issue and ongoing discussion - https://github.com/projectdiscovery/nuclei/issues/491, We will be adding more variables, not all but a few important ones like you mentioned above.

```
{{Port}}
{{Proto}}
```

**Reply by Techbrunch** [reply]
Great, in the meantime it would be nice to list the one that are available in the documentation.

**Reply by ehsandeep** [reply]
It's here - https://nuclei.projectdiscovery.io/templating-guide/protocols/http/

![image](https://user-images.githubusercontent.com/8293321/118699577-51c87f00-b82f-11eb-8a5d-6356ef6c0832.png)
